### PR TITLE
fix(macros): Broken Specifications macros fixed by escaping or adding spec URLs

### PR DESCRIPTION
**Pull request by bsmth**
This PR fixes instances of broken Specifications macros. 

* Escaping instances of usage in a template
* Adding `spec_urls` in guides where appropriate

### CODE REVIEW

**Comment by yin1999** [comment]
A note here, two spec macro calls have been removed in #27147.

File: files/en-us/mdn/writing_guidelines/page_structures/page_types/aria_page_template/index.md [hunk 1]
```diff
@@ -122,7 +122,7 @@ Any additional benefits this feature has for non-typical screen reader users, li
 
 ## Specifications
 
-{{Specifications}}
+\\{{Specifications}}
```
Comment (dipikabh) [inline files/en-us/mdn/writing_guidelines/page_structures/page_types/aria_page_template/index.md:125]: The way this is done in the [CSS property page template](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Page_types/CSS_property_page_template#specifications) is:

```
## Specifications

`\{{Specifications}}`

_Remember to remove the backticks and backslash to use this macro._

```
Should we do it the same way here?

(Also, out of scope here, but maybe a follow up PR would be needed to fix this section in all templates. I could take that on.)

**Review by dipikabh — CHANGES_REQUESTED** [review]
Hi @bsmth, thanks for fixing these! Just one comment / suggestion

File: files/en-us/mdn/writing_guidelines/page_structures/page_types/aria_page_template/index.md (hunk 1 above)
Comment (bsmth) [inline files/en-us/mdn/writing_guidelines/page_structures/page_types/aria_page_template/index.md:125]:
```suggestion
`\{{Specifications}}`

_Remember to remove the backticks and backslash to use this macro._
```

Comment (bsmth) [reply files/en-us/mdn/writing_guidelines/page_structures/page_types/aria_page_template/index.md]: Good call, done in https://github.com/mdn/content/pull/27113/commits/4fe04d3baaede5899e617d6c86222f6cd1c3c966

**Review by dipikabh — APPROVED** [review]
Looks good, thank you 👍

**Pull request merged**